### PR TITLE
Fix accepted types for fputcsv fields parameter

### PR DESCRIPTION
**Pull request by DaveLiddament**
Correctly fix type for `fputcsv` function parameter `$fields`. Fix attempt made in #448 

- [X] Update code
- [X] Add tests
- [x] Fix failing test

### CODE REVIEW

File: tests/PHPStan/Rules/Functions/CallToFunctionParametersRuleTest.php [hunk 1]
```diff
@@ -411,6 +411,28 @@ public function testUnpackOperator(): void
...
+				33,
+			],
+		]);
+	}
+
+
+	public function testPutCsvWithStringable(): void
+	{
+		if (PHP_VERSION_ID < 80000) {
+			$this->markTestSkipped('Test skipped under 8.0');
```
Comment (ondrejmirtes) [inline tests/PHPStan/Rules/Functions/CallToFunctionParametersRuleTest.php:428]: I read "under 8.0" as "when running 8.0". Something like "before" or "on lower version than" would be nicer.

File: tests/PHPStan/Rules/Functions/data/fputcsv-fields-parameter-php8.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?php declare(strict_types = 1);
+
+class StringablePerson implements \Stringable {
```
Comment (ondrejmirtes) [inline tests/PHPStan/Rules/Functions/data/fputcsv-fields-parameter-php8.php:3]: Please don't put these test files in global namespace. Some namespace like "namespace FputcsvPhp8" would work.

File: tests/PHPStan/Rules/Functions/data/fputcsv-fields-parameter.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<?php declare(strict_types=1);
+
+class Person
```
Comment (ondrejmirtes) [inline tests/PHPStan/Rules/Functions/data/fputcsv-fields-parameter.php:3]: Please don't put these test files in global namespace. Some namespace like "namespace FputcsvTest" would work.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: tests/PHPStan/Rules/Functions/data/fputcsv-fields-parameter.php (hunk 3 above)
Comment (DaveLiddament) [reply tests/PHPStan/Rules/Functions/data/fputcsv-fields-parameter.php:5]: Added

File: tests/PHPStan/Rules/Functions/data/fputcsv-fields-parameter-php8.php (hunk 2 above)
Comment (DaveLiddament) [reply tests/PHPStan/Rules/Functions/data/fputcsv-fields-parameter-php8.php:5]: Added

File: tests/PHPStan/Rules/Functions/CallToFunctionParametersRuleTest.php (hunk 1 above)
Comment (DaveLiddament) [reply tests/PHPStan/Rules/Functions/CallToFunctionParametersRuleTest.php]: Added a better message.

**Review by ondrejmirtes — APPROVED** [review]

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**